services/ui_backend_service/data/cache/get_card_data_action.py

```diff
@@ -0,0 +1,86 @@
+from typing import Callable, List
+
+from services.ui_backend_service.data.cache.utils import streamed_errors
+
+from .get_data_action import GetData
+
+from metaflow import Task
+from metaflow.cards import get_cards
+import os
+
+
+class GetCardData(GetData):
+    @classmethod
+    def format_request(cls, pathspec: str, card_hash: str, invalidate_cache=False):
+        """
+        Cache Action to fetch Cards HTML content for a pathspec
+
+        Parameters
+        ----------
+        pathspec : str
+            Task pathspec:
+                ["FlowId/RunNumber/StepName/TaskId"]
+        card_hash : str
+            Card hash
+        """
+        targets = "{}/{}".format(pathspec, card_hash)
+        # We set the targets to have the card hash so that we can find the card data for a particular hash
+        return super().format_request(
+            targets=[targets], invalidate_cache=invalidate_cache
+        )
+
+    @classmethod
+    def fetch_data(
+        cls, pathspec_with_card_hash: str, stream_output: Callable[[str], None]
+    ):
+        """
+        Fetch data using Metaflow Client.
+
+        Parameters
+        ----------
+        pathspec_with_card_hash : str
+            Task pathspec with card hash
+                "FlowId/RunNumber/StepName/TaskId/CardHash"
+        stream_output : Callable[[object], None]
+            Stream output callable from execute() that accepts a JSON serializable object.
+            Used for generic messaging.
+
+        Errors can be streamed to cache client using `stream_output` in combination with
+        the error_event_msg helper. This way failures won't be cached for individual artifacts,
+        thus making it necessary to retry fetching during next attempt.
+        (Will add significant overhead/delay).
+
+        Stream error example:
+            stream_output(error_event_msg(str(ex), "s3-not-found", get_traceback_str()))
+        """
+
+        def _card_item(card):
+            card_load_policy = os.environ.get("MF_CARD_LOAD_POLICY", "full")
+            card_data = None
+            if card_load_policy == "full":
+                card_data = card.get_data()
```

Comment:
`.get_data` being added in https://github.com/Netflix/metaflow/pull/1550 so this service PR should only go in once the client changes are released. We also need to pin the Metaflow client version to a new minimum for the service release

Reply:
Will add a dependency pin!